kitsune/sumo/middleware.py

```diff
@@ -199,8 +219,12 @@ def safe_query_string(request):
 # This is a modified version of 'mobility.middleware.DetectMobileMiddleware'.
 # We want to exclude tablets from being detected as MOBILE and there is
 # no way to do that by just overriding the detection regex.
-class DetectMobileMiddleware(object):
+class DetectMobileMiddleware(MiddlewareMixin):
     """Looks at user agent and decides whether the device is mobile."""
+    def __init__(self, *args, **kwargs):
+        if settings.SKIP_MOBILE_DETECTION:
```

Comment:
I think we should do the same for [`MobileSwitchMiddleware`](https://github.com/mozilla/kitsune/pull/3140/files#diff-6f05e342c77cc4322987a6ff3c3edd64R241) which will allow users to switch to mobile by passing `?mobile=1` in the URL

Reply:
Is there a reason to disallow that for the CDN though? It doesn't rely on headers we'd have to worry about and is technically a different URL if it has different parameters.

Reply:
Also in my searching it looks like a lot of code expects `request.MOBILE` to be set, and the `MobileSwitchMiddleware` is the only thing I see that sets it. I believe a lot of things would break if it were disabled. We could have it just set `request.MOBILE = False` and skip the other functions in READ_ONLY if that'd be better.

Reply:
> Is there a reason to disallow that for the CDN though? It doesn't rely on headers we'd have to worry about and is technically a different URL if it has different parameters.

My concern is that if someone visits with `mobile=1` the cache would store the mobile version of the page?

Reply:
We cache (and we pass) no querystrings atm

![image](https://user-images.githubusercontent.com/584352/37770018-aa09be86-2ddb-11e8-9ece-f328e0fad0cd.png)

Since we're not passing them to the app I guess we can leave this middleware as is or hardcode to `False` as you suggest. What do you think?

Reply:
huh... does no other part of the site rely on query strings? I believe searching does right? e.g.: 

https://support.mozilla.org/en-US/search/advanced?q=insecure&language=en-US&sortby_documents=relevance&a=1&w=1

I think we need to turn that back on actually.